src/main/java/com/thealgorithms/datastructures/lists/CircleDoublyLinkedList.java

```diff
@@ -0,0 +1,356 @@
+package com.thealgorithms.datastructures.lists;
+
+public class CircleDoublyLinkedList<E> {
+
+    private class Node<T> {
```

Comment:
Take a look at the classes under [devutils.nodes](https://github.com/TheAlgorithms/Java/tree/master/src/main/java/com/thealgorithms/devutils/nodes) and see if you can use one of those instead of creating a private class.